pusher/src/Controller/AuthenticateController.ts

```diff
@@ -41,24 +35,22 @@ export class AuthenticateController extends BaseController {
                     playUri as string | undefined,
                     redirect as string | undefined
                 );
-                res.writeStatus("302");
-                res.writeHeader("Location", loginUri);
-                return res.end();
+                res.status(302);
+                res.setHeader("Location", loginUri);
+                return res;
             } catch (e) {
                 console.error("openIDLogin => e", e);
-                return this.errorToResponse(e, res);
+                this.castErrorToResponse(e, res);
+                return;
             }
         });
     }
 
     openIDCallback() {
         //eslint-disable-next-line @typescript-eslint/no-misused-promises
-        this.App.get("/login-callback", async (res: HttpResponse, req: HttpRequest) => {
-            res.onAborted(() => {
-                console.warn("/message request was aborted");
-            });
-            const IPAddress = req.getHeader("x-forwarded-for");
-            const { code, nonce, token, playUri } = parse(req.getQuery());
+        this.app.get("/login-callback", async (req, res) => {
+            const IPAddress = req.header("x-forwarded-for");
+            const { code, nonce, token, playUri } = parse(req.path_query);
```

Comment:
Can we simplify this further to avoid calling "parse" ?

Reply:
It's the only way I've found to not have a "as any"